images/orbit-controller/src/orbit_controller/operators/orbitjob_operator.py

```diff
@@ -167,3 +167,54 @@ def create_job(
         "orbitJobOperator": {"jobStatus": "JobCreated", "jobName": job_instance_metadata.name, "nodeType": node_type}
     }
     return "JobCreated"
+
+
+@kopf.on.timer(
+    ORBIT_API_GROUP,
+    ORBIT_API_VERSION,
+    "orbitjobs",
+    interval=5,
+    initial_delay=20
+    # when=_should_index_jobs
+)
+def orbit_job_monitor(
+    namespace: str,
+    name: str,
+    patch: kopf.Patch,
+    logger: kopf.Logger,
+    namespaces_idx: kopf.Index[str, Dict[str, Any]],
+    jobs_idx: kopf.Index[Tuple[str, str], Dict[str, Any]],
+    **_: Any,
+) -> Any:
+    ns: Optional[Dict[str, Any]] = None
+    k8s_job: Optional[Dict[str, Any]] = None
+
+    for ns in namespaces_idx.get(namespace, []):
+        logger.debug("ns: %s", ns)
+
+    if ns is None:
+        patch["status"] = {
+            "orbitJobOperator": {"jobStatus": "JobDetailsNotFound", "error": "No Namespace resource found"}
+        }
+        return "JobDetailsNotFound"
+
+    for k8s_job in jobs_idx.get((namespace, name), []):
+        logger.debug("k8s_job: %s", k8s_job)
+
+    if k8s_job is None:
```

Comment:
without a filter, when the Job completes and the TTL deletes it, this will fire and clear the status on the OrbitJob.